### PR TITLE
[ROSTESTS] Fix crash in winhttp:winhttp regression test

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Improve wine_dbgstr_wn function by partial sync to Wine-6.14

_The winhttp:winhttp test was often crashing on wine_dbgstr_wn._

JIRA issue: [ROSTESTS-377](https://jira.reactos.org/browse/ROSTESTS-377)

## Sync our test.h file's wine_dbgstr_wn as much as possible to Wine's

_This brings most of Wine's current wine_dbgstr_wn code into ReactOS._
It is not possible to completely sync this to current Wine.
This is because there is Wine-specific code regarding "debug_info" and their TEB in remaining code.
I confirmed this with @ThFabba.

### CODE REVIEW

File: sdk/include/reactos/wine/test.h [hunk 1]
```diff
@@ -634,30 +634,24 @@ const char *wine_dbgstr_an( const CHAR *str, intptr_t n )
     return res;
 }
 
-const char *wine_dbgstr_wn( const WCHAR *str, intptr_t n )
+const char *wine_dbgstr_wn( const WCHAR *str, int n )
```
Comment (HBelusca) [inline sdk/include/reactos/wine/test.h:637]: No idea what Wine drank, but this should certainly be `intptr_t` or `size_t` for x64 compliance.

Comment (Doug-Lyons) [reply sdk/include/reactos/wine/test.h]: Changed. Thanks.

File: sdk/include/reactos/wine/test.h [hunk 2]
```diff
@@ -668,14 +662,15 @@ const char *wine_dbgstr_wn( const WCHAR *str, intptr_t n )
...
-            else
+            if (c < ' ' || c >= 127)
             {
                 *dst++ = '\\';
-                sprintf(dst,"%04x",c);
-                dst+=4;
+                *dst++ = hex[(c >> 12) & 0x0f];
+                *dst++ = hex[(c >> 8) & 0x0f];
+                *dst++ = hex[(c >> 4) & 0x0f];
+                *dst++ = hex[c & 0x0f];
```
Comment (learn-more) [inline sdk/include/reactos/wine/test.h:671]: This code is written for a 4-bytes WCHAR, which we don't have.
So the upper 2 digits will always be 0.

Comment (Doug-Lyons) [reply sdk/include/reactos/wine/test.h:671]: Mark, Can you please review https://reactos.org/testman/diff.php?id1=56015182&id2=56024828&type=1&strip=0 and look at line #4279 on the left and the right and see if it makes sense to remove any of these lines? Thanks. Doug Lyons

Comment (learn-more) [reply sdk/include/reactos/wine/test.h:671]: Sorry, I misread the code.
It is good as-is.

**Review by learn-more — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
Thanks for your help in getting this merged, @binarymaster. We will have to watch the winhttp:winhttp test for a while and see what it shows. I do have another patch in [ROSTESTS-377](https://jira.reactos.org/browse/ROSTESTS-377) that may be needed.

**Pull request merged**